git_hooks/pre-push

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+# Called by "git push" after it has checked the remote status, but before anything has been
+# pushed.  If this script exits with a non-zero status nothing will be pushed.
+#
+# This hook is called with the following parameters:
+#
+# $1 -- Name of the remote to which the push is being done
+# $2 -- URL to which the push is being done
+#
+# If pushing without using a named remote those arguments will be equal.
+#
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <local ref> <local sha1> <remote ref> <remote sha1>
+#
+set -eu
+
+REMOTE="$1"
+URL="$2"
+
+echo "REMOTE is $REMOTE"
+echo "URL is $URL"
+
+./tests/scripts/check-doxy-blocks.pl >/dev/null
+./tests/scripts/check-names.sh >/dev/null
+./tests/scripts/check-generated-files.sh >/dev/null
```

Comment:
Thanks for considering my previous comments. While this script is simpler than the previous versions, I still have a few comments:

* The code `command >/dev/null` only redirects the stdout output to `/dev/null`, but does not handle stderr.
* I think I might be contradicting a little of what I commented on previously. I advised to redirect the output to `/dev/null`. However, disposing of the output when there is a failure is a problem because then the user has no idea what actually failed, so he/she would have to open the hooks script and manually try every test command until something failed. I still think that the output of successful tests should be disposed, but more indications regarding the cause of the failure should given. For example, the command that failed and the output?